controller/server.go

```diff
@@ -44,12 +44,19 @@ func (rw *ReadWrite) TaskCreate(ctx context.Context, taskConfig config.TaskConfi
 	d.SetBufferPeriod()
 
 	// Add the task driver to the driver list only after successful create
-	err = rw.drivers.Add(*taskConfig.Name, d)
+	name := *taskConfig.Name
+	err = rw.drivers.Add(name, d)
 	if err != nil {
 		return config.TaskConfig{}, err
```

Comment:
I know adding a driver is not likely to error, but should we technically also be cleaning up here too?

Reply:
yeah that's a good point, I'll add cleanup here as well